falcon/middlewares.py

```diff
@@ -1,16 +1,101 @@
+from typing import Iterable, Optional, Union
+
+from .request import Request
+from .response import Response
+
 
 class CORSMiddleware(object):
-    def process_response(self, req, resp, resource, req_succeeded):
-        """Implement a simple blanket CORS policy for all routes.
+    """CORS Middleware
+
+    This middleware provides a simple out-of-the box CORS policy, including handling
+    of preflighted requests from the browser.
+
+    See also:
+
+    * https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS
+    * https://www.w3.org/TR/cors/#resource-processing-model
+
+    Keyword Arguments:
+        allow_origins (Union[str, Iterable[str]]): List of origins to allow (case
+            sensitive). The string ``'*'`` acts as a wildcard, matching every origin.
+            (default ``'*'``).
+        expose_headers (Optional[Union[str, Iterable[str]]]): List of additional headers
+            to expose via the ``Access-Control-Expose-Headers`` header.
+            These headers are in addition to the CORS-safelisted ones:
+            ``Cache-Control``, ``Content-Language``, ``Content-Length``, ``Content-Type``,
+            ``Expires``, ``Last-Modified``, ``Pragma``. (default ``None``).
+
+            See also:
+            https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Expose-Headers
+        allow_credentials (Optional[Union[str, Iterable[str]]]): List of origins (case sensitive)
+            for which to allow credentials via the ``Access-Control-Allow-Credentials`` header.
+            The string ``'*'`` acts as a wildcard, matching every all allowed origins. Set to
```

Comment:
> matching every all allowed origins

Something got mixed up here, :arrow_right: _matching every ~~all~~ allowed origin_ or _matching ~~every~~ all allowed origins_

Reply:
must have forgot to re-read that sentence. I've gone with every